ethpm_types/utils.py

```diff
@@ -1,61 +1,33 @@
 from enum import Enum
 from hashlib import md5, sha3_256, sha256
-from typing import Tuple, Union
+from typing import Tuple
 
 from hexbytes import HexBytes as BaseHexBytes
+from pydantic import PlainSerializer, PlainValidator, WithJsonSchema
+from pydantic_core import core_schema
 
-from ethpm_types._pydantic_v1 import FileUrl, _AnyUrl
+try:
+    from typing import Annotated
+except ImportError:
+    from typing_extensions import Annotated  # type: ignore
 
-CONTENT_ADDRESSED_SCHEMES = {"ipfs"}
-AnyUrl = Union[_AnyUrl, FileUrl]
-
-
-class HexBytes(BaseHexBytes):
-    """
-    A class representing bytes as a hex-str.
-    """
-
-    @classmethod
-    def __get_validators__(cls):
-        yield cls.validate
-
-    @classmethod
-    def validate(cls, v):
-        return HexBytes(v)
-
-
-class Algorithm(str, Enum):
-    """
-    Algorithm enum options MD5, SHA3, and SHA256.
-    """
 
-    MD5 = "md5"
-    SHA3 = "sha3"
-    SHA256 = "sha256"
-
-
-def is_valid_hex(data: str) -> bool:
-    """
-    Returns ``True`` if the given data is a valid hex str.
-    """
-
-    if not data.startswith("0x"):
-        return False
-
-    if set(data[2:].lower()) > set("1234567890abcdef"):
-        return False
+CONTENT_ADDRESSED_SCHEMES = {"ipfs"}
 
-    if len(data) % 2 != 0:
-        return False
 
-    return True
+HexBytes = Annotated[
```

Comment:
This is awesome

Reply:
Another thing I kinda like:

```python
AddressType = Annotated[ChecksumAddress, BeforeValidator(validate_address)]
```

^ This will make it so `AddressType` is _always_ converted when used in Pydantic models.
So we can delete the field validator `def validate_address(cls, value)` that we use and redefine in many places.

I am also thinking about it if it would make sense to move `AddressType` to `ethpm-types` so that even `evm-trace` can have this be an automatic thing.

Reply:
It also incidentally fixes a doc linking problem with `AddressType` in Ape

Reply:
> I am also thinking about it if it would make sense to move `AddressType` to `ethpm-types` so that even `evm-trace` can have this be an automatic thing.

Game for this

Reply:
with the invention of `eth-pydantic-types`, we no longer are doing this. Now, the most basic types can easily be shared and it just makes more sense. Evm trace no longer needs ethpm-types etc.